src/builtincontext.cc

```diff
@@ -6,27 +6,31 @@
 #include "printutils.h"
 #include "evalcontext.h"
 
-BuiltinContext::BuiltinContext()
+BuiltinContext::BuiltinContext() : Context()
 {
-	for(const auto &ass : Builtins::instance()->getAssignments()) {
-		this->set_variable(ass.name, ass.expr->evaluate(this));
+}
+
+void BuiltinContext::init()
```

Comment:
Not sure I understand why initialization moved from the constructor to `init()`. Feels a bit counter to how I'm used to think about C++ RAII.

Reply:
Passing `this` to evaluate as `shared_ptr` does not work in the constructor. I think to fix the issue we will need to decouple the evaluation functions from the context objects. Same issue also in Module and Function context which want to pass `this` along.

Reply:
Ah, right,  the `shared_from_this()` hack.  That's  a bit scary, but  since we always construct context  objects  held by shared_ptrs, we should be safe.